### PR TITLE
Readme update

### PR DESCRIPTION
**Pull request by britt-mo**
## Description
More Detail to the README for devs to have an understanding of how the app is structured, how forms are generated, programs are saved/stored, and other more specific information around functionality. WIP so any and all suggestions are welcome.

## Code author checklist
- [ ] I have performed a self-review of my code
- [ ] I have added [thorough](https://bit.ly/3zPrxuZ) tests
- [ ] I have added analytics, if necessary
- [ ] I have updated the documentation, if necessary

## Reviewer checklist (two different people)
- [ ] I have done the deep review and verified the items in the above checklist are g2g
- [ ] I have done the lgtm review

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -222,6 +222,41 @@ When a pull request is approved and merged into main the deploy script will spin
 
 ![Architecture Diagram](./.images/architecture.svg?raw=true)
 
+```text
```
Comment (BearHanded) [inline README.md:225]: This shouldn't be an individual text block, and the below items can probably be filtered into appropriate homes in the rest of the readme.

File: README.md [hunk 2]
```diff
@@ -222,6 +222,41 @@ When a pull request is approved and merged into main the deploy script will spin
...
+
+**File in the bucket, vs high level db entry for a form** - When a user creates a form, it is stored in Dynamo and tracks user information such as when the program was last edited and by whom, date submitted’ report period start and end date, program name, report type, the state, id, and status. The file in the S3 bucket is the entire form of user inputted data, and this is a pattern that is unique to this project. S3 is mainly used for attachments, data for virus scans on attachments, mathematica integration. We decided to store the programs in S3 because these data can get so large that we can’t reliably store it all in Dynamo, nor search through them without the app breaking.
```
Comment (BearHanded) [inline README.md:258]: Maybe "Dynamo macpar-reports vs macpar-form in S3 Storage"

File: README.md [hunk 3]
```diff
@@ -222,6 +222,41 @@ When a pull request is approved and merged into main the deploy script will spin
...
+
+**Custom JSON & form field creation engine (formFieldFactory)** - each report has a custom JSON object, stored in a JSON file, written using a custom schema. This JSON object is referred to as the form template and it is the blueprint from which report form fields are created. It is also used to create routes and navigation elements throughout the app. When provided form fields from this template, the formFieldFactory renders the appropriate form fields. A similar process occurs when a report is exported in PDF preview format. 
+
+**Page and Form Structure** Each page has a name, path, and pageType, for example the first page a user sees in the form will be have ‘pageType: standard’ with a ‘verbiage’ object that includes all of the text that precedes the form fields. The the ‘form’ object follows with a unique id and ‘fields’ array that holds one or more objects that represent the individual questions in a form. There are different types of forms as well. If there is a "pageType": "modalDrawer", then instead of a ‘form’ object, it will have a ‘modalForm’ object.
```
Comment (BearHanded) [inline README.md:237]: Can we at least get an example of a basic page template, with its corresponding field answers, and anything confusing explained?

**Review by BearHanded — CHANGES_REQUESTED** [review]

File: README.md (hunk 3 above)
Comment (britt-mo) [reply README.md]: Good feedback, I'll implement.

File: README.md [hunk 4]
```diff
@@ -8,6 +8,14 @@ MDCT-MCR is an application meant to collect state report data to help the health
 
 Managed Care is a health care delivery system organized to manage cost, utilization, and quality. Medicaid managed care provides for the delivery of Medicaid health benefits and additional services through contractual arrangements between state Medicaid agencies and Managed Care Organizations (MCOs) that accept a set per member per month (capitation) payment for these services.
 
+There are three reports that users complete:
+
+**MCPAR** - (Managed Care Program Annual Report) report for states to complete annually for each Medicaid managed care program
+
+**MLR** - (Medical Loss Ratio) report for states to complete annually or when contract terms change
+
+**NAAAR** -
```
Comment (tblackwe) [inline README.md:17]: NAAAR - (Network Adequacy and Access Assurances Report) report for states to complete annually, when contract terms change, or there is a significant change in operations

**Review by tblackwe — APPROVED** [review]

**Review by karla-vm — APPROVED** [review]

**Pull request merged**